src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -571,7 +571,7 @@ class CreateFunctionStatementSegment(BaseSegment):
         Ref("FunctionNameIdentifierSegment"),
         "AS",
         Ref("SingleOrDoubleQuotedLiteralGrammar"),
-        Ref("ResourceLocationGrammar"),
+        Ref("ResourceLocationGrammar", optional=True),
```

Comment:
Nit: this seems unrelated to the PR and there's no mention of it in the PR description. So can you confirm that this is intentional?

(In the future it's best to keep PRs focussed on the stated goal or at least call out these small fixes)

Reply:
yes, I'll make an additional note in the top-level PR comment.  It's intentional as this should not be required syntax

Reply:
awesome thanks 👍